api/src/test/java/org/pmiops/workbench/api/DataSetControllerTest.java

```diff
@@ -0,0 +1,419 @@
+    dataSet.addValuesItem(valueSet);
+    ValueSet valueSetTwo = new ValueSet().domain(Domain.DRUG).values(new DomainValuesResponse().items(domainValues));
+    dataSet.addValuesItem(valueSetTwo);
+
+
+    List<String> selectConditionStrings = new ArrayList<>();
+    selectConditionStrings.add("PERSON_ID");
+    List<String> joinConditionStrings = new ArrayList<>();
+    joinConditionStrings.add("FROM `all-of-us-ehr-dev.synthetic_cdr20180606.condition_occurrence` c_occurrence");
+
+    List<String> selectDrugStrings = new ArrayList<>();
+    selectDrugStrings.add("PERSON_ID");
+    List<String> joinDrugStrings = new ArrayList<>();
+    joinDrugStrings.add("FROM `all-of-us-ehr-dev.synthetic_cdr20180606.drug_exposure` d_exposure");
+
+    dataSetController = spy(dataSetController);
+    doReturn(new ValuesLinkingPair(selectConditionStrings, joinConditionStrings)).when(dataSetController).getValueSelectsAndJoins(valueSet, Domain.CONDITION);
+    doReturn(new ValuesLinkingPair(selectDrugStrings, joinDrugStrings)).when(dataSetController).getValueSelectsAndJoins(valueSetTwo, Domain.DRUG);
+
+    DataSetQueryResponse response = dataSetController.getQueryFromDataSet(WORKSPACE_NAMESPACE, WORKSPACE_NAME, dataSet).getBody();
+    assertThat(response.getQueryList()).isNotEmpty();
+    verify(dataSetController, times(2)).getValueSelectsAndJoins(any(), any());
+    assertThat(response.getQueryList().size()).isEqualTo(2);
+  }
+
+  @Test
+  public void testGetQueryTwoCohorts() {
+    DataSet dataSet = buildEmptyDataSet();
+    dataSet = dataSet.addCohortIdsItem(COHORT_ONE_ID);
+    dataSet = dataSet.addCohortIdsItem(COHORT_TWO_ID);
+    dataSet = dataSet.addConceptSetIdsItem(CONCEPT_SET_ONE_ID);
+    List<DomainValue> domainValues = new ArrayList<>();
+    domainValues.add(new DomainValue().value("PERSON_ID"));
+    ValueSet valueSet = new ValueSet().domain(Domain.CONDITION).values(new DomainValuesResponse().items(domainValues));
+    dataSet.addValuesItem(valueSet);
+
+
+    List<String> selectStrings = new ArrayList<>();
+    selectStrings.add("PERSON_ID");
+    List<String> joinStrings = new ArrayList<>();
+    joinStrings.add("FROM `all-of-us-ehr-dev.synthetic_cdr20180606.condition_occurrence` c_occurrence");
+
+    dataSetController = spy(dataSetController);
+    doReturn(new ValuesLinkingPair(selectStrings, joinStrings)).when(dataSetController).getValueSelectsAndJoins(valueSet, Domain.CONDITION);
+
+    DataSetQueryResponse response = dataSetController.getQueryFromDataSet(WORKSPACE_NAMESPACE, WORKSPACE_NAME, dataSet).getBody();
+    assertThat(response.getQueryList().size()).isEqualTo(1);
+    verify(dataSetController, times(1)).getValueSelectsAndJoins(valueSet, Domain.CONDITION);
+    assertThat(response.getQueryList().get(0).getQuery()).matches("SELECT(.*)WHERE CONDITION_CONCEPT_ID IN \\((.*)\\)(.*)PERSON_ID IN \\(SELECT(.*)OR PERSON_ID IN(.*)");
```

Comment:
Similar comments to the above for this assertion. If the only thing we care about checking is that we're generating two "PERSON_ID in" strings, it might be simpler to check that the string matches twice against the substring "PERSON_ID in".

Alternatively, it might not be bad to make the test slightly more brittle, in order to better document the output SQL.

Reply:
I think I want to test that it has `UNION PERSON_ID in` specifically. The rest of the query structure we may not need to test here, because it should be caught by the above.